WordPress/Sniffs/Theme/NoDeregisterCoreScriptSniff.php

```diff
@@ -0,0 +1,165 @@
+			'suggest'                  => 'suggest',
+			'thickbox'                 => 'thickbox',
+			'hoverIntent'              => 'hoverIntent',
+			'jquery-hotkeys'           => 'jquery-hotkeys',
+			'sack'                     => 'sack',
+			'quicktags'                => 'quicktags',
+			'iris'                     => 'iris',
+			'farbtastic'               => 'farbtastic',
+			'colorpicker'              => 'colorpicker',
+			'tiny_mce'                 => 'tiny_mce',
+			'autosave'                 => 'autosave',
+			'wp-ajax-response'         => 'wp-ajax-response',
+			'wp-lists'                 => 'wp-lists',
+			'common'                   => 'common',
+			'editorremov'              => 'editorremov',
+			'editor-functions'         => 'editor-functions',
+			'ajaxcat'                  => 'ajaxcat',
+			'admin-categories'         => 'admin-categories',
+			'admin-tags'               => 'admin-tags',
+			'admin-custom-fields'      => 'admin-custom-fields',
+			'password-strength-meter'  => 'password-strength-meter',
+			'admin-comments'           => 'admin-comments',
+			'admin-users'              => 'admin-users',
+			'admin-forms'              => 'admin-forms',
+			'xfn'                      => 'xfn',
+			'upload'                   => 'upload',
+			'postbox'                  => 'postbox',
+			'slug'                     => 'slug',
+			'post'                     => 'post',
+			'page'                     => 'page',
+			'link'                     => 'link',
+			'comment'                  => 'comment',
+			'comment-reply'            => 'comment-reply',
+			'admin-gallery'            => 'admin-gallery',
+			'media-upload'             => 'media-upload',
+			'admin-widgets'            => 'admin-widgets',
+			'word-count'               => 'word-count',
+			'theme-preview'            => 'theme-preview',
+			'json2'                    => 'json2',
+			'plupload'                 => 'plupload',
+			'plupload-all'             => 'plupload-all',
+			'plupload-html4'           => 'plupload-html4',
+			'plupload-html5'           => 'plupload-html5',
+			'plupload-flash'           => 'plupload-flash',
+			'plupload-silverlight'     => 'plupload-silverlight',
+			'underscore'               => 'underscore',
+			'backbone'                 => 'backbone',
+			'imagesloaded'             => 'imagesloaded',
+		),
+	);
+
+	/**
+	 * Process the parameters of a matched function.
+	 *
+	 * @since 0.xx.0
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 *
+	 * @return void
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+
+		if ( ! isset( $parameters[1] ) ) {
+			return;
+		}
+
+		$matched_parameter = $this->strip_quotes( $parameters[1]['raw'] );
```

Comment:
Just using the `raw` value makes this an extremely easy sniff to bypass:
```php
wp_deregister_script( 'j' . 'query' );
```
Better to walk the tokens within the parameter if there is more than one token (discounting whitespace and comments).